File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = collections.defaultdict(list)
        
        for str in strs:
            sorted_str = ''.join(sorted(str))
            result[sorted_str].append(str)
        
        return result.values()
            

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> group;
        for (const auto &s: strs) {
            string sortedS = s;
            sort(sortedS.begin(), sortedS.end());
            group[sortedS].push_back(s);
        }
        vector<vector<string>> res;
        for (auto &pair: group) {
            res.push_back(pair.second);
        }
        return res;
    }
};